src/osd/PG.cc

```diff
@@ -8558,7 +8558,7 @@ boost::statechart::result PG::RecoveryState::Active::react(const MNotifyRec& not
 		       << dendl;
     pg->proc_replica_info(
       notevt.from, notevt.notify.info, notevt.notify.epoch_sent);
-    if (pg->have_unfound()) {
+    if (pg->have_unfound() || pg->is_degraded()) {
```

Comment:
It looks like in `PG::proc_replica_info()` we are inserting the `from` osd correctly in `might_have_unfound` https://github.com/ceph/ceph/blob/master/src/osd/PG.cc#L495, so just checking if this osd is present in this set, instead of `pg->is_degraded()` should be enough.

Reply:
makes sense. i've updated the patch :)